app/models/members/group_member.rb

```diff
@@ -4,9 +4,22 @@ module Members
   # entity class for GroupMember
   class GroupMember < Member
     belongs_to :group, foreign_key: :namespace_id # rubocop:disable Rails/InverseOf
+    before_destroy :last_namespace_owner_member
 
     def self.sti_name
       'GroupMember'
     end
+
+    # Method to ensure we don't leave a group or project without an owner
+    def last_namespace_owner_member
```

Comment:
This still needs a bit more work.

I think you should first check that the member being removed is an owner and then you can use the statement where you check for many

Reply:
Updated to return if destroyed by association, if the member isn't an owner, and only run the query to check for many if the first 2 statements are not true in a11c9c768378fa5e50e557466b32a1b73e43342f